src/openrct2/ride/Ride.cpp

```diff
@@ -397,8 +397,23 @@ money32 Ride::CalculateIncomePerHour() const
 
     if (currentShopItem != SHOP_ITEM_NONE)
     {
-        priceMinusCost += price[1];
-        priceMinusCost -= ShopItems[currentShopItem].Cost;
+        money16 photoProfit = price[1] - ShopItems[currentShopItem].Cost;
+
+        if (lifecycle_flags & RIDE_LIFECYCLE_ON_RIDE_PHOTO)
```

Comment:
I prefer this if to check for `shop_item_is_photo`.

Reply:
What do you mean? Do I change it?

Reply:
Yeah, use `shop_item_is_photo(currentShopItem)` instead of the current check.

Reply:
@Gymnasiast I've updated the code.